### PR TITLE
Fetch classic Menus and Pages using view (readonly) context

### PR DESCRIPTION
**Pull request by getdave**
~This PR builds on https://github.com/WordPress/gutenberg/pull/37685 and acts as a test case for that PR.~ #37685 is now merged so we can proceed here if ✅ 

## Description

On `trunk` we are currently attempting to "read" data about Menus and Pages but we are using the `edit` context to do so. Therefore when you are a lower permission user (e.g. `Contributor` role) [this results in a 403 error](https://github.com/WordPress/gutenberg/pull/37685#issuecomment-1009987522) because the user doesn't have `edit` permissions.

This PR follows what @youknowriad suggests which is to adjust the `context` parameter on the call to `getEntityRecords` to be `view`. This means the request will succeed and the data can be read.

Note this test case is rather limited.

## How has this been tested?

- Create Contributor user.
- Switch to Contributor user.
- New Post.
- Add Nav block
- See no 403 for either Menus or Pages endpoints.

Compare this to `trunk` where you will see 403 errors.

Note that the UI won't change because only `Admin` role users are allowed to create Menus and thus access to creating Navigation Menus from either Classic Menus or Pages is not exposed in the UI.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/core-data/src/entities.js
Comment (getdave) [inline packages/core-data/src/entities.js]: FYI these are the changes from https://github.com/WordPress/gutenberg/pull/37685

**Comment by spacedmonkey** [comment]
Screenshot 2022-01-11 at 17 37 55

I am not seeing any error in the network tab, which is good. But the dropdown is empty here. Am I missing something here?

**Comment by getdave** [comment]
> But the dropdown is empty here. Am I missing something here?

That's expected. It's easy to miss. Here's what I wrote in the PR description:

> Note that the UI won't change because only Admin role users are allowed to create Menus and thus access to creating Navigation Menus from either Classic Menus or Pages is not exposed in the UI.

**Comment by spacedmonkey** [comment]
> That's expected. It's easy to miss. Here's what I wrote in the PR description:

Okay, if that is the case, then the dropdown should be hidden or something. A empty dropdown looks weird / broken.

**Comment by getdave** [comment]
> > That's expected. It's easy to miss. Here's what I wrote in the PR description:
> 
> Okay, if that is the case, then the dropdown should be hidden or something. A empty dropdown looks weird / broken.

Oh I see. Looking again this morning with a less tired brain I see that's  a bug. The `Menus` heading within the dropdown deals with block based menus. As you say it shouldn't show if there are either

- no block based Navigation Menus
- no Classic Menus

**Comment by spacedmonkey** [comment]
The perfect solution here, would be, if you are a contributor, show a list of classic menus and allow you to create a navigation block, but save the data to the post content.  

What is the point of making the api request and to be able to see a list of classic menus and not be able to do anything about it...

**Comment by getdave** [comment]
> The perfect solution here, would be, if you are a contributor, show a list of classic menus and allow you to create a navigation block, but save the data to the post content.

I agree that we may need to introduce a mechanic that allows us to save Nav blocks to the post content rather than the CPT. This will be required for example, when we start building more complex "mega menus" that may not be suitable to be saved without presentational attributes and thus won't be designed to be transferable across Themes.

> What is the point of making the api request and to be able to see a list of classic menus and not be able to do anything about it...

Well indeed. Ideally we want to avoid unnecessary API requests and it should be possible to do so.

**Comment by getdave** [comment]
I've [addressed the UI issue of showing/hiding the `Select menu` option in the placeholder](https://github.com/WordPress/gutenberg/pull/37980) based on various criteria here. Note this _doesn't_ avoid the API requests being fired. That will require additional work.

**Comment by spacedmonkey** [comment]
I have approved #37980. Once this is merge and this PR is rebased, I will re-review this.

**Comment by getdave** [comment]
Bear in mind we cannot merge this until @youknowriad merges https://github.com/WordPress/gutenberg/pull/37685

**Comment by getdave** [comment]
> I have approved #37980. Once this is merge and this PR is rebased, I will re-review this.

@spacedmonkey This PR has now been rebased.

**Comment by spacedmonkey** [comment]
Once a navigation block is create by an admin, contributors can edit it.

**Comment by spacedmonkey** [comment]
This PR looks good. It is a nicpic, not sure it should be part of this PR. But the is a random line here after navigation. 
Screenshot 2022-01-17 at 14 46 57

Otherwise this PR looks good.

**Comment by getdave** [comment]
> But the is a random line here after navigation.

Yeh fair enough. It's not conditionally shown because - I guess - we never intended such a state to exist. Indeed I wonder whether we ought to have some inline message here? Seems like a confusing state.

If not then I'd suggest we raise a _follow up_ PR to only show his `<hr />` _if_ there are no items.

**Comment by spacedmonkey** [comment]
I have created a follow up ticket, for allow contributor's to create navigation blocks. See https://github.com/WordPress/gutenberg/issues/38030

**Review by spacedmonkey — APPROVED** [review]

**Comment by getdave** [comment]
@spacedmonkey did you want this backported for the RC?

**Comment by spacedmonkey** [comment]
@getdave i do think it should be yes

**Pull request merged**
